src/colp/components/usetype_version_comparison_report.py

```diff
@@ -0,0 +1,78 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+import plotly.express as px
+from st_aggrid import AgGrid
+from src.constants import COLOR_SCHEME
+from abc import ABC
+import pdb
+
+
+class UsetypeVersionComparisonReport:
+    def __init__(self, data):
+        self.usetype_changes = data["usetype_changes"]
+
+    def __call__(self):
+        st.subheader("Usetype Version Comparison")
+
+        if self.usetype_changes is None:
+            st.info("There is no usetype version comparison report for this branch.")
+            return
+
+        self.version = st.sidebar.selectbox(
+            "Select a Version for Comparison", self.usetype_changes.v_current.unique()
```

Comment:
The issue was that the hardcoded date here was already out of date - thought that would be more stable, for this solution I moved this sidebar back into this class since it now was sufficiently tied to the logic of the report for it to make sense together - we could think through a different solution if you like though